documentation/puppet-api/v3/task_detail.markdown

```diff
@@ -0,0 +1,196 @@
+---
+layout: default
+title: "Puppet Server: Puppet API: Task detail"
+canonical: "/puppetserver/latest/puppet-api/v3/task-detail.html"
+---
+
+[deprecated WEBrick Puppet master]: https://docs.puppet.com/puppet/latest/reference/services_master_webrick.html
+[`environment_timeout`]: https://docs.puppet.com/puppet/latest/reference/config_file_environment.html#environmenttimeout
+
+[`auth.conf`]: ../../config_file_auth.markdown
+[`puppetserver.conf`]: ../../config_file_puppetserver.markdown
+
+The tasks API provides access to task information stored in modules. Tasks are
+files stored in `tasks` subdirectory of a module. A task consists of an
+executable file, with an optional metadata file with the same name with an
+added '.json' extension. For example, the "install" task in a module "apache" could
+consist of the executable file `install.rb` and the metadata file
+`install.json`. This task would have the display name "apache::install".
```

Comment:
Was there discussion on how this could be confusing if there was also a manifest called `install` (install.pp) in the apache module?

Reply:
It's an example, I just used "apache" because that seems to be a commonly-used example module. I don't know the name of any planned tasks/modules.